src/main/java/com/geronimo/model/AuditedEntity.java

```diff
@@ -0,0 +1,25 @@
+package com.geronimo.model;
+
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.SequenceGenerator;
+import javax.validation.constraints.NotNull;
+import java.util.Date;
+
+@Setter
+@ToString
+@MappedSuperclass
+abstract class AuditedEntity {
```

Comment:
Spring Data has a special support for auduting: https://docs.spring.io/spring-data/data-jpa/docs/2.0.x/reference/html/#auditing

But it's better to have it configured like this: https://docs.spring.io/spring-data/data-jpa/docs/2.0.x/reference/html/#jpa.auditing to not have a vendor dependency on entity layer